script/sync_prod_to_gcp/subscribe_submissions.py

```diff
@@ -0,0 +1,213 @@
+
+    subscriber_client = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber_client.subscription_path(project_id, subscription_id)
+    callback = test_callback if test else submission_callback
+    streaming_pull_future = subscriber_client.subscribe(subscription_path, callback=callback)
+    with subscriber_client:
+        try:
+            streaming_pull_future.result()
+        except Exception as e:
+            logger.error("Subscribe failed: %s", str(e), exc_info=True)
+            streaming_pull_future.cancel()
+
+
+if __name__ == "__main__":
+    ad = argparse.ArgumentParser(epilog=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    ad.add_argument('--project',
+                    help='GCP project name. Default is arxiv-production',
+                    dest="project", default="arxiv-production")
+    ad.add_argument('--topic',
+                    help='GCP pub/sub name. The default matches with arxiv-production',
+                    dest="topic", default="submission-published")
+    ad.add_argument('--subscription',
+                    help='Subscription name. Default is the one in production', dest="subscription",
+                    default="sync-submission-from-cit-to-gcp")
+    ad.add_argument('--json-log-dir',
+                    help='JSON logging directory. The default is correct on the sync-node',
+                    default='/var/log/e-prints')
+    ad.add_argument('--debug', help='Set logging to debug. Does not invoke testing',
+                    action='store_true')
+    ad.add_argument('--test', help='Test reading the queue but not do anything',
+                    action='store_true')
+    ad.add_argument('--bucket',
+                    help='The bucket name. The default is mentioned in sync_published_to_gcp so for the production, you do not need to provide this - IOW it automatically uses the same buchet as sync-to-gcp',
+                    dest="bucket", default="")
+    args = ad.parse_args()
+
+    project_id = args.project
+    subscription_id = args.subscription
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+
+    if args.json_log_dir and os.path.exists(args.json_log_dir):
+        json_logHandler = logging.handlers.RotatingFileHandler(os.path.join(args.json_log_dir, "submissions.log"),
+                                                               maxBytes=4 * 1024 * 1024,
+                                                               backupCount=10)
+        json_formatter = ArxivSyncJsonFormatter(**LOG_FORMAT_KWARGS)
+        json_formatter.converter = gmtime
+        json_logHandler.setFormatter(json_formatter)
+        json_logHandler.setLevel(logging.DEBUG if args.debug else logging.INFO)
+        logger.addHandler(json_logHandler)
+        pass
+
+    # Ensure the environment is authenticated
+    if "GOOGLE_APPLICATION_CREDENTIALS" not in os.environ:
+        logger.error("Environment variable 'GOOGLE_APPLICATION_CREDENTIALS' is not set.")
```

Comment:
Do we really want to error out here?
In particular, when running in dev mode, we could rely on the default login credentials from `gcloud login`, which would not require downloading credentials, or copying somewhere very specific.
Every user with access rights could run it immediately.

Reply:
The script is intended to run on CIT sync-node. Only way to get there is to use the VPN. Since other cron jobs do the same, singling this out doesn't make sense.